### PR TITLE
test: use kyverno/chainsaw for E2E testing

### PR DESCRIPTION
**Pull request by odubajDT**
Fixes: #623 

## Changes

- used kyverno-/chainsaw for e2e testing
- created separate WF for e2e testing
- adapted tests to be executed by kyverno/chainsaw

### Advantages

- better logging when failures
- more built-in features for testing (wait, assert, cleanups, applying,...)
- faster execution
- newer technology

### CODE REVIEW

File: .chainsaw.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/kyverno/chainsaw/main/.schemas/json/configuration-chainsaw-v1alpha1.json
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Configuration
+metadata:
+  name: configuration
+spec:
+  skipDelete: true
+  template: true
+  timeouts:
+    assert: 6m40s
```
Comment (thisthat) [inline .chainsaw.yaml:10]: [q] how did you come up with these magic numbers? are they based on the current CI time + some margin?

Comment (odubajDT) [reply .chainsaw.yaml:10]: to be honest I just copied them now, I will do some polishing and proper descriptions of the changes in the next day (due to lack of time)

File: .chainsaw-local.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/kyverno/chainsaw/main/.schemas/json/configuration-chainsaw-v1alpha1.json
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Configuration
+metadata:
+  name: configuration-local
+spec:
+  template: true
+  timeouts:
+    assert: 2m
```
Comment (odubajDT) [inline .chainsaw-local.yaml:9]: 4x the usual execution time for the tests (~30s)

File: .github/workflows/pr-checks.yml [hunk 3]
```diff
@@ -97,62 +95,13 @@ jobs:
...
-      - name: Upload cluster logs
-        if: failure()
-        uses: actions/upload-artifact@v3
-        with:
-          name: e2e-tests
-          path: .github/scripts/logs
+        kind_tag: [ v1.22.17, v1.23.17, v1.24.15, v1.25.11, v1.26.3, v1.27.3, v1.28.0 ]
+    with:
+      kind_tag: ${{ matrix.kind_tag }}
+    uses: ./.github/workflows/e2e.yml
```
Comment (odubajDT) [inline .github/workflows/pr-checks.yml:107]: created a separate WF for the e2e tests

File: test/e2e/chainsaw/assets/manifests.yaml [hunk 4]
```diff
@@ -87,7 +87,9 @@ spec:
 apiVersion: batch/v1
 kind: Job
 metadata:
-  name: flagd-query-test
+  name: (join('-', [$namespace, 'job1']))
```
Comment (odubajDT) [inline test/e2e/chainsaw/assets/manifests.yaml:90]: clearly distinguish the jobs in the cluster

**Review by thisthat — APPROVED** [review]
LGTM :) We should merge this after https://github.com/open-feature/open-feature-operator/pull/643/files#diff-10ab0d0a4c64d29f58046c064328b7f77ad9d9571a463ce0530647b464b1217e since it also touches some kuttl e2e test

**Review by bacherfl — APPROVED** [review]

**Review by toddbaert — APPROVED** [review]

**Pull request merged**